### PR TITLE
fix: improve csv name handling

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash-internal-work/issues/1297
Closes: https://github.com/lightdash/lightdash-internal-work/issues/1298

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/pages/ProjectSettings.tsx [hunk 1]
```diff
@@ -103,9 +103,7 @@ const ProjectSettings: FC = () => {
                         <SettingsValidator projectUuid={projectUuid} />
                     </Route>
 
-                    <Redirect
-                        to={`/generalSettings/projectManagement/${projectUuid}/settings`}
-                    />
+                    <Redirect to={`/generalSettings/`} />
```
Comment (rephus) [inline packages/frontend/src/pages/ProjectSettings.tsx:106]: Should fix 

> .fortknight/repos/portal_lightdash/packages/frontend/src/components/SettingsValidator/ValidatorTable/index.tsx

File: packages/frontend/src/components/SettingsValidator/ValidatorTable/index.tsx [hunk 2]
```diff
@@ -97,26 +121,11 @@ const TableValidationItem = forwardRef<
     const { hovered, ref: isHoveredRef } = useHover<HTMLTableRowElement>();
 
     return (
-        <tr
-            ref={mergeRefs(ref, isHoveredRef)}
-            onClick={() =>
```
Comment (rephus) [inline packages/frontend/src/components/SettingsValidator/ValidatorTable/index.tsx:102]: Should fix: 

> .fortknight/repos/portal_lightdash/packages/frontend/src/components/SettingsValidator/ValidatorTable/index.tsx

I replaced this with another anchor

File: packages/frontend/src/components/NavBar/SettingsMenu.tsx [hunk 3]
```diff
@@ -57,7 +57,7 @@ const SettingsMenu: FC = () => {
                     <Menu.Item
                         component={Link}
                         icon={<MantineIcon icon={IconDatabase} />}
-                        to={`/generalSettings/projectManagement/${activeProjectUuid}`}
+                        to={`/generalSettings/projectManagement/${activeProjectUuid}/settings`}
```
Comment (rephus) [inline packages/frontend/src/components/NavBar/SettingsMenu.tsx:60]: Required after removing redirect on settings

File: packages/backend/src/routers/slackRouter.ts [hunk 4]
```diff
@@ -59,15 +60,12 @@ slackRouter.get(
     async (req, res, next) => {
         try {
             const { imageId } = req.params;
-            if (
-                !imageId.startsWith('slack-image') ||
-                !imageId.endsWith('.png')
-            ) {
+            if (!UnfurlService.isValidImageFileId(imageId)) {
```
Comment (rephus) [inline packages/backend/src/routers/slackRouter.ts:63]: Should fix 

> .fortknight/repos/portal_lightdash/packages/backend/src/routers/slackRouter.ts

File: packages/backend/src/routers/slackRouter.ts [hunk 5]
```diff
@@ -59,15 +60,12 @@ slackRouter.get(
...
                 throw new NotFoundError(
                     `Slack image not found ${req.params.imageId}`,
                 );
             }
-            const sanitizedImageId = imageId.replace('..', '');
+            const sanitizedImageId = path.normalize(imageId);
```
Comment (rephus) [inline packages/backend/src/routers/slackRouter.ts:68]: Should fix

 >    .fortknight/repos/portal_lightdash/packages/backend/src/routers/slackRouter.ts

File: packages/backend/src/routers/projectRouter.ts [hunk 6]
```diff
@@ -229,12 +229,12 @@ projectRouter.get(
         try {
             const { fileId } = req.params;
 
-            if (!fileId.startsWith('csv-') || !fileId.endsWith('.csv')) {
+            if (!CsvService.isValidCsvFileId(fileId)) {
                 throw new NotFoundError(`CSV file not found ${fileId}`);
             }
-            const sanitizedFileId = fileId.replace('..', '');
-
+            const sanitizedFileId = path.normalize(fileId);
```
Comment (rephus) [inline packages/backend/src/routers/projectRouter.ts:235]: Should fix 

> .fortknight/repos/portal_lightdash/packages/backend/src/routers/projectRouter.ts

**Review by owlas — APPROVED** [review]

**Pull request merged**